crates/sound/src/lib.rs

```diff
@@ -224,16 +223,39 @@ impl SoundConfig {
     }
 }
 
-pub type SoundBitmap = ();
-
-#[derive(Debug)]
-pub struct SoundRequest<'a> {
-    data_slice: Option<VolatileSlice<'a, SoundBitmap>>,
+pub struct IOMessage {
+    status: std::sync::atomic::AtomicU32,
+    desc_chain: SoundDescriptorChain,
+    descriptor: virtio_queue::Descriptor,
+    vring: VringRwLock,
 }
 
-impl<'a> SoundRequest<'a> {
-    pub fn data_slice(&self) -> Option<&VolatileSlice<'a, SoundBitmap>> {
-        self.data_slice.as_ref()
+impl Drop for IOMessage {
+    fn drop(&mut self) {
+        log::trace!("dropping IOMessage");
+        let resp = VirtioSoundPcmStatus {
+            status: self.status.load(std::sync::atomic::Ordering::SeqCst).into(),
+            latency_bytes: 0.into(),
+        };
+
+        if let Err(err) = self
+            .desc_chain
+            .memory()
+            .write_obj(resp, self.descriptor.addr())
+        {
+            log::error!("Error::DescriptorWriteFailed: {}", err);
+            return;
+        }
+        if self
+            .vring
+            .add_used(self.desc_chain.head_index(), resp.as_slice().len() as u32)
+            .is_err()
+        {
+            log::error!("Couldn't add used");
+        }
+        if self.vring.signal_used_queue().is_err() {
+            log::error!("Couldn't signal used queue");
+        }
```

Comment:
The main drawback of this approach (which I don't dislike) is that we don't return an error via `process_*()`, right?

Reply:
Indeed, I took this approach from crates/gpio/src/vhu_gpio.rs and I'm not sure what to do here. I'll have to check what the vhost-user backend trait does with the errors returned by the handling code